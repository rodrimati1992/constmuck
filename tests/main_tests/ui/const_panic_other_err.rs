// different size
const _: u8 = constmuck::pod_read_unaligned(&[0u8; 2]);
const _: u32 = constmuck::pod_read_unaligned(&[0u8; 2]);

#[cfg(feature = "debug_checks")]
const _: () = {
    #[derive(Copy, Clone)]
    #[repr(transparent)]
    struct Foo(u8);

    unsafe impl constmuck::Contiguous for Foo {
        type Int = u8;
        const MIN_VALUE: u8 = 3;
        const MAX_VALUE: u8 = 2;
    }

    let _: Option<Foo> = constmuck::contiguous::from_integer(4);
};






fn main(){}