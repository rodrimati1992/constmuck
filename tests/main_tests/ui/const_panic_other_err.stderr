error[E0080]: evaluation of constant value failed
 --> tests/main_tests/ui/const_panic_other_err.rs:2:15
  |
2 | const _: u8 = constmuck::pod_read_unaligned(&[0u8; 2]);
  |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at '
the sizes of `T` and the slice are not the same
slice length: 2
size_of::<T>(): 1', $DIR/tests/main_tests/ui/const_panic_other_err.rs:2:15

error[E0080]: evaluation of constant value failed
 --> tests/main_tests/ui/const_panic_other_err.rs:3:16
  |
3 | const _: u32 = constmuck::pod_read_unaligned(&[0u8; 2]);
  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at '
the sizes of `T` and the slice are not the same
slice length: 2
size_of::<T>(): 4', $DIR/tests/main_tests/ui/const_panic_other_err.rs:3:16

error[E0080]: evaluation of constant value failed
  --> tests/main_tests/ui/const_panic_other_err.rs:17:26
   |
17 |     let _: Option<Foo> = constmuck::contiguous::from_integer(4);
   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at '
`T` implements `Contiguous` where `T::MIN_VALUE` is larger than `T::MAX_VALUE`
T::MIN_VALUE: 3
T::MAX_VALUE: 2', $DIR/tests/main_tests/ui/const_panic_other_err.rs:17:26
